tool/bazelinstall/rbe.sh

```diff
@@ -2,18 +2,21 @@
 
 set -e
 
-if [[ -n "$BAZEL_CACHE_URL" ]]; then
+if [[ -n "$BAZEL_CACHE_CREDENTIAL" ]] && [[ -n "$BAZEL_CACHE_URL" ]]; then
     echo "Installing remote cache credential..."
-    BAZEL_CACHE_CREDENTIAL=/opt/credentials/
-    echo "A remote cache credential is found and will be saved to $BAZEL_CACHE_CREDENTIAL. Artifact and test results will be cached remotely."
-    sudo mkdir -p $BAZEL_CACHE_CREDENTIAL && sudo chmod a+rwx $BAZEL_CACHE_CREDENTIAL
-    ( echo "build --remote_cache=$BAZEL_CACHE_URL"; echo "test --remote_cache=$BAZEL_CACHE_URL"; echo "run --remote_cache=$BAZEL_CACHE_URL" ) >> "$BAZEL_CACHE_CREDENTIAL/bazel-remote-cache.rc"
-    echo "The RBE credential has been installed!"
-    echo "Configuring Python..."
-    pyenv global system system
+    BAZEL_CACHE_CREDENTIAL_LOCATION=./credential.json
+    echo "A remote cache credential is found and will be saved to $BAZEL_CACHE_CREDENTIAL_LOCATION. Artifact and test results will be cached remotely."
+    BAZEL_CACHE_CREDENTIAL_JSON=$(echo "$BAZEL_CACHE_CREDENTIAL" | base64 -d)
+    if [[ -n "$BAZEL_CACHE_CREDENTIAL_JSON" ]]; then
+      echo "$BAZEL_CACHE_CREDENTIAL_JSON" > $BAZEL_CACHE_CREDENTIAL_LOCATION
+      ( echo "build --remote_cache=$BAZEL_CACHE_URL --google_credentials=$BAZEL_CACHE_CREDENTIAL_LOCATION"; echo "test --cache_test_results=no" ) > ./.bazel-remote-cache.rc
+      echo "The remote cache has been installed!"
+    else
+      echo "The remote cache was not installed."
+      echo "\$BAZEL_CACHE_CREDENTIAL was non-empty, but failed to be decoded as base64. Confirm your value is valid."
```

Comment:
Would it be more in line with our engineering practices to give a non-zero exit code here? To get here, this script has to be called explicitly and the cache credential and cache URL environment variables have to be set, but the cache credential failed deserialisation from base64. 

As it currently stands, the user could think they've installed the cache but actually failed to. One would have to read their Factory/Bazel logs carefully to find out that this is the case.

Reply:
I've gone ahead and implemented this. Let me know if this is undesirable.